src/test/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScoreTest.java

```diff
@@ -239,12 +244,141 @@ public void testFourteenMonthOldSingleVersion() {
+    assertEquals(6.0, value.get(), DELTA);
+
+    final Optional<Value<?>> versionsValue =
+        value.usedValues().stream().filter(v -> v instanceof ArtifactVersionsValue).findFirst();
+    assertTrue(versionsValue.isPresent());
+    assertTrue(((ArtifactVersions) versionsValue.get().get()).containsAllOf(artifactVersions));
+  }
+
+  @Test
+  public void testVersionsSelectionIfArtifactIsNonSemantic() {
+    LocalDateTime now = LocalDateTime.now();
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", now.minusDays(6 * 31));
+    ArtifactVersion nonSematicVersion = new ArtifactVersion("MIGHTY-1.0", now.minusDays(4 * 31));
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(3 * 31));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(2 * 31));
+
+    ArtifactVersions artifactVersions =
+        ArtifactVersions.of(version100, nonSematicVersion, version110, version200);
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    ScoreValue value = score.calculate(RELEASED_ARTIFACT_VERSIONS.value(artifactVersions),
+        ARTIFACT_VERSION.value(nonSematicVersion));
+    assertEquals(7.3, value.get(), DELTA);
+
+    final Optional<Value<?>> versionsValue =
+        value.usedValues().stream().filter(v -> v instanceof ArtifactVersionsValue).findFirst();
+    assertTrue(versionsValue.isPresent());
+    assertTrue(((ArtifactVersions) versionsValue.get().get()).containsAllOf(artifactVersions));
+  }
+
+  @Test
+  public void testBasicFunctionalityWithUnknownArtifactVersions() {
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", LocalDateTime.now().minusMonths(14));
+
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    ScoreValue value =
+        score.calculate(RELEASED_ARTIFACT_VERSIONS.unknown(), ARTIFACT_VERSION.value(version100));
+    assertTrue(value.isUnknown());
+  }
+
+  @Test
+  public void testBasicFunctionalityWithUnknownValues() {
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    ScoreValue value =
+        score.calculate(RELEASED_ARTIFACT_VERSIONS.unknown(), ARTIFACT_VERSION.unknown());
+    assertTrue(value.isUnknown());
+  }
+
+  @Test
+  public void testBasicFunctionalityWithUnknownArtifactVersion() {
+    LocalDateTime now = LocalDateTime.now();
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", now.minusDays(6 * 31));
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(4 * 31));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(2 * 31));
+
+    ArtifactVersions artifactVersions = ArtifactVersions.of(version100, version110, version200);
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    ScoreValue value = score.calculate(RELEASED_ARTIFACT_VERSIONS.value(artifactVersions),
+        ARTIFACT_VERSION.unknown());
+    assertEquals(5.0, value.get(), DELTA);
```

Comment:
It would be good to check if confidence is not max here. Same for other test cases that use unknown values.